src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -349,12 +355,72 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
         trigger=TriggerUpdateParameters(
             source_triggers=source_trigger_update_params,
             base_image_trigger=base_image_trigger_update_params
+        ),
+        credentials=get_custom_registry_credentials(
+            cmd,
+            auth_mode=auth_mode
+        )
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+
+
+def acr_task_credential_add(cmd,
+                            client,
+                            task_name,
+                            registry_name,
+                            login_server,
+                            username,
+                            password,
+                            resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    TaskUpdateParameters = cmd.get_models('TaskUpdateParameters')
+    taskUpdateParameters = TaskUpdateParameters(
+        credentials=get_custom_registry_credentials(
+            cmd,
+            login_server=login_server,
+            username=username,
+            password=password
         )
     )
 
     return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
 
 
+def acr_task_credential_delete(cmd,
+                               client,
+                               task_name,
+                               registry_name,
+                               login_server,
+                               resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    TaskUpdateParameters = cmd.get_models('TaskUpdateParameters')
+    taskUpdateParameters = TaskUpdateParameters(
+        credentials=get_custom_registry_credentials(
+            cmd,
+            login_server=login_server
+        )
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
```

Comment:
The add/delete methods are returning a different obj than the list method, is this by design?

Reply:
Yes, the `client.update` method in this case returns a `msrest.polling.poller.LROPoller` object. I cannot extract out `credentials` from that.

While `list` returns a `azure.mgmt.containerregistry.v2018_09_01.models.task_py3.Task` object.

Reply:
cc @mnltejaswini @northtyphoon